mocks/sender.go

```diff
@@ -0,0 +1,88 @@
+package mocks
+
+import (
+	"context"
+	"sync"
+
+	pb "github.com/topfreegames/protos/eventsgateway/grpc/generated"
+)
+
+type MockSender struct {
+	totalSent           int
+	totalCalls          int
+	failureIndexesOrder [][]int64
+	firstCallEvents     []*pb.Event
+	eventsIdsFreqs      map[string]int
+	mutex               *sync.Mutex
```

Comment:
It's usual Go style to call your mutex `mu` and place it above the fields it's supposed to protect. Please either do that or comment in the mutex which fields it's meant to guard.